web/webkit/src/test/scala/net/liftweb/http/FutureWithSessionSpec.scala

```diff
@@ -0,0 +1,272 @@
+package net.liftweb.http
+
+import net.liftweb.common.Empty
+import scala.concurrent._
+import org.specs2.mutable.Specification
+import FutureWithSession._
+
+import scala.util.{Try, Success}
+
+object FutureWithSessionSpec extends Specification {
+
+  object TestVar1 extends SessionVar[String]("Uninitialized1")
+  object TestVar2 extends SessionVar[String]("Uninitialized2")
+
+  "A FutureWithSession" should {
+    "fail if session is not available" in {
+      val actual:Future[String] = Future("something").withCurrentSession
+
+      // TODO: Why the hell does this not work??
+//      val expected:Option[Try[String]] = Some(Failure(new Exception))
+//      actual.value must eventually(beEqualTo(expected))
+
+      actual.value.isDefined must eventually(beTrue)
+      actual.value.get must beFailedTry.withThrowable[Exception]("LiftSession not available in this thread context")
```

Comment:
I think you can avoid `get`s and such here (and clear Codacy's PR annoyances) by using `beLike`:

```
actual.value must eventually(beLike {
  case Some(Failure(exception)) =>
    exception must beInstanceOf[exception]
})
```